src/par_iter/chain.rs

```diff
@@ -0,0 +1,173 @@
+use super::*;
+use super::internal::*;
+use std::cmp::min;
+use std::iter;
+
+pub struct ChainIter<A, B>
+    where A: ParallelIterator, B: ParallelIterator<Item=A::Item>
+{
+    a: A,
+    b: B,
+}
+
+impl<A, B> ChainIter<A, B>
+    where A: ParallelIterator, B: ParallelIterator<Item=A::Item>
+{
+    pub fn new(a: A, b: B) -> ChainIter<A, B> {
+        ChainIter { a: a, b: b }
+    }
+}
+
+impl<A, B> ParallelIterator for ChainIter<A, B>
+    where A: ParallelIterator, B: ParallelIterator<Item=A::Item>
+{
+    type Item = A::Item;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let a = self.a.drive_unindexed(consumer.split_off());
```

Comment:
I am always [scared of segfaults](https://users.rust-lang.org/t/rust-guarantees-no-segfaults-with-only-safe-code-but-it-segfaults-stack-overflow/4305) when I see recursion...